Fixing issue #27009  – Continue search functionality in MembersListCard

arup-chauhan opened the pull request:
Hi team,

This follows up on the earlier efforts in PR #29471 and PR #30833 for the issue #27009, which began implementing this feature but were not merged.

### Changes in this PR

* Added a search bar (`TextField`) to filter members in `MembersListCard`.
* Expanded filtering to include `metadata.name`, `metadata.title`, `spec.profile?.displayName`, and `spec.profile?.email`.
* Reset pagination to page 1 when the search term changes.
* Updated member count display to reflect filtered results (`Members (x of y)`).
* Added test coverage for:

  * Filtering by name, title, display name, and email.
  * “No members” state when search yields no results.

### Screenshots

Attached 3 screenshots:  — with 2 results, with 1 result,  without results* (old ones moved below)

**Jan Update:** Added spacing between the search field and members list to address layout feedback, along with 3 updated screenshots.

<img width="622" height="380" alt="Jan 1 Member Search" src="https://github.com/user-attachments/assets/7893f175-3fb7-4169-8378-a7ac2e9a7aa5" />
<img width="610" height="528" alt="Jan 2 Member Search" src="https://github.com/user-attachments/assets/d9554424-b0fe-4728-8422-d347dc43431f" />
<img width="615" height="381" alt="Jan No Member Search" src="https://github.com/user-attachments/assets/34b8bb05-5250-4779-8b16-45bfed54c3f2" />

Test results: 

<img width="621" height="151" alt="Backstage Test Passes" src="https://github.com/user-attachments/assets/f585cca0-0a80-47aa-939d-0b7b388ae2c8" />

### Checklist
* [X] Verified UI in example app.
* [X] All tests passing (`yarn test`).
* [ ] Updated documentation if necessary.
<img width="849" height="406" alt="Backstage 1 Member Found" src="https://github.com/user-attachments/assets/b12a3737-4293-4fa3-a16e-dedb0a8da85d" />
<img width="714" height="520" alt="Backstage 2 Member Found" src="https://github.com/user-attachments/assets/bb8a0c49-d1ff-4267-99cc-cf3c1beaa6c1" />
<img width="1232" height="433" alt="Backstage No Member Found" src="https://github.com/user-attachments/assets/4988277e-6923-4c2a-9c0f-b2f66f85e0fd" />

benjdlambert commented:
@arup-chauhan I think this needs a rebase from main. There's some changesets that appear to have been deleted here which shouldn't be in this PR?

I'd expect only the changes scoped to the org plugin, and just a single changeset, but it looks like there's too much going on in this PR and branch? :pray:

arup-chauhan commented:
> @arup-chauhan I think this needs a rebase from main. There's some changesets that appear to have been deleted here which shouldn't be in this PR?
> 
> I'd expect only the changes scoped to the org plugin, and just a single changeset, but it looks like there's too much going on in this PR and branch? 🙏

Hey @benjdlambert 

I rebased on upstream master, restored the upstream changeset folder, removed the stray file, and added a single patch changeset scoped to plugin org. The diff is now clean and only includes the member search functionality and tests. Ready for review.

benjdlambert commented on plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx:
```diff
@@ -308,7 +331,18 @@ export const MembersListCard = (props: {
         ) : (
-          memberList
+          <>
+            <TextField
```
Does these need some padding to the members underneath? 

In the design it looks a little tight?
